src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs

```diff
@@ -520,22 +502,48 @@ public async Task<decimal> GetWalletBalance()
 
         public void OnDemandPayoutAsync()
         {
-            messageBus.Listen<NetworkBlockNotification>().Subscribe(b =>
+            messageBus.Listen<NetworkBlockNotification>().Subscribe(block =>
             {
-                logger.Info($"[{LogCategory}] NetworkBlockNotification height={b.BlockHeight}, gasfee={b.BaseFeePerGas}");
+                logger.Info($"[{LogCategory}] NetworkBlockNotification height={block.BlockHeight}, gasfee={block.BaseFeePerGas}");
+
+                // Handle an invalid gas fee
+                if (block.BaseFeePerGas <= 0)
+                {
+                    logger.Warn($"[{LogCategory}] NetworkBlockNotification invalid gas fee value, gasfee={block.BaseFeePerGas}");
+                    return;
+                }
+
+                // Check if we are over our hard limit
+                if (block.BaseFeePerGas > extraConfig.MaxGasLimit)
+                {
+                    logger.Info($"[{LogCategory}] Gas exceeds the MaxGasLimit={extraConfig.MaxGasLimit}, skipping payouts");
+                    return;
+                }
 
-                if(b.BaseFeePerGas <= 0) return;
+                // The default minimum payout is based on the configuration
+                var minimumPayout = poolConfig.PaymentProcessing.MinimumPayment;
 
-                if(b.BaseFeePerGas > (extraConfig.MaxGasLimit * extraConfig.TopMinersGasLimitFactor)) return;
+                // If the GasDeductionPercentage is set, use a ratio instead
+                if (extraConfig.GasDeductionPercentage > 0)
+                {
+                    var latestGasFee = block.BaseFeePerGas / EthereumConstants.Wei;
+                    if (latestGasFee <= 0)
+                    {
+                        throw new Exception($"LatestGasFee is invalid: {latestGasFee}");
+                    }
+                    var transactionCost = (21000 * latestGasFee);
```

Comment:
instead of hard coding 21000, can you use 
EthereumPoolPaymentProcessingConfigExtra.Gas?

that should also be set to 21000, but from config, in case another fork comes along that changes it.

Reply:
Fixed